src/pudl/extract/ferc1.py

```diff
@@ -49,21 +49,110 @@
 and EIA 923.
 
 """
+import csv
+import importlib
 import logging
-import os.path
-import re
-import string
+import zipfile
+from pathlib import Path
 
 import dbfread
 import pandas as pd
 import sqlalchemy as sa
+from dbfread import DBF
 
 import pudl
-import pudl.constants as pc
-import pudl.workspace.datastore as datastore
+from pudl import constants as pc
+from pudl.workspace import datastore as datastore
 
 logger = logging.getLogger(__name__)
 
+MIN_RID = 1
+MAX_RID = 531
+PUDL_RIDS = {
+    514: "AEP Texas",
+    519: "Upper Michigan Energy Resources Company",
+    522: "Luning Energy Holdings LLC, Invenergy Investments",
+    529: "Tri-State Generation and Transmission Association",
+    531: "Basin Electric Power Cooperative",
+}
+
+
+def missing_respondents(min_rid, max_rid, existing_rids, pudl_rids):
+    """Fill in missing respondents for the f1_respondent_id table."""
+    records = []
+    for rid in range(min_rid, max_rid + 1):
+        if rid in existing_rids:
+            continue
+        elif rid in pudl_rids:
+            records.append(
+                {
+                    "respondent_id": rid,
+                    "respondent_name": f"{pudl_rids[rid]} (PUDL determined)"
+                },
+            )
+        else:
+            records.append(
+                {
+                    "respondent_id": rid,
+                    "respondent_name": f"Missing Respondent {rid}"
+                },
+            )
+    return records
+
+
+class Ferc1Datastore(datastore.Datastore):
+    """Provide a thin interface for pulling files from the Datastore."""
+
+    def get_dir(self, year):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=6339827)